pkg/component/kubernetes/apiserverexposure/service_test.go

```diff
@@ -184,44 +187,55 @@ var _ = Describe("#Service", func() {
-			It("should successfully deploy with expected kube-apiserver service annotations and labels", func() {
-				defaultDepWaiter = NewService(
-					log,
-					c,
-					namespace,
-					&ServiceValues{
-						AnnotationsFunc:             func() map[string]string { return map[string]string{"foo": "bar"} },
-						NamePrefix:                  namePrefix,
-						TopologyAwareRoutingEnabled: true,
-						RuntimeKubernetesVersion:    semver.MustParse("1.31.1"),
-					},
-					func() client.ObjectKey { return sniServiceObjKey },
-					&retryfake.Ops{MaxAttempts: 1},
-					clusterIPsFunc,
-					ingressIPFunc,
-				)
+	Context("when TopologyAwareRoutingEnabled=true", func() {
+		BeforeEach(func() {
+			values.TopologyAwareRoutingEnabled = true
+		})
 
+		When("runtime Kubernetes version is >= 1.32", func() {
+			BeforeEach(func() {
+				values.RuntimeKubernetesVersion = semver.MustParse("1.32.1")
+			})
+
+			It("should successfully deploy with expected kube-apiserver service annotation, label and spec field", func() {
 				Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
 
 				actual := &corev1.Service{}
 				Expect(c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: expectedName}, actual)).To(Succeed())
 
-				expected.Annotations = map[string]string{
-					"foo":                          "bar",
-					"networking.istio.io/exportTo": "*",
-					"networking.resources.gardener.cloud/from-all-garden-scrape-targets-allowed-ports": `[{"protocol":"TCP","port":443}]`,
-					"networking.resources.gardener.cloud/namespace-selectors":                          `[{"matchLabels":{"gardener.cloud/role":"istio-ingress"}},{"matchLabels":{"networking.gardener.cloud/access-target-apiserver":"allowed"}}]`,
-					"service.kubernetes.io/topology-mode":                                              "auto",
-				}
-				expected.Labels = map[string]string{
-					"app": "kubernetes",
-					"endpoint-slice-hints.resources.gardener.cloud/consider": "true",
-					"role": "apiserver",
-				}
-				expected.Spec.Type = corev1.ServiceTypeClusterIP
-				Expect(actual).To(DeepEqual(expected))
+				Expect(actual.Spec.TrafficDistribution).To(PointTo(Equal(corev1.ServiceTrafficDistributionPreferClose)))
+			})
+		})
+
+		When("runtime Kubernetes version is 1.31", func() {
+			BeforeEach(func() {
+				values.RuntimeKubernetesVersion = semver.MustParse("1.31.1")
+			})
+
+			It("should successfully deploy with expected kube-apiserver service annotation, label and spec field", func() {
+				Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
+
+				actual := &corev1.Service{}
+				Expect(c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: expectedName}, actual)).To(Succeed())
+
+				Expect(actual.Spec.TrafficDistribution).To(PointTo(Equal(corev1.ServiceTrafficDistributionPreferClose)))
+				Expect(actual.Labels).To(HaveKeyWithValue("endpoint-slice-hints.resources.gardener.cloud/consider", "true"))
+			})
+		})
+
+		Context("when K8s version < 1.31", func() {
```

Comment:
Use `When` to make it consistent to other tests or change the ones above to use `Context`.